c++/src/vector.cc

```diff
@@ -44,18 +45,26 @@ class Vector {
     return this->dataStore_[length_--];
   }
 
-  unsigned int length() const{
-    return this->length_;
-  }
+  void reserve(unsigned int reserveIndices) {
+    while (this->capacity_ < reserveIndices) {
+      this->capacity_ = this->capacity_ * this->growFactor;
+    }
```

Comment:
You with your math phd might be able to elide this loop. Like maybe log2(reserveIndices), if it's a perfect integer then just `this->capacity = reserveIndices`, otherwise `this->capacity = std::pow(2, power + 1)`.

For example if `reserveIndices == 128`, log2(128) = 7, so `this->capacity = reserveIndices`. If `reserveIndices == 129`, log2(129) = 7.011, so `this->capacity = std::pow(2, 7 + 1)`.

This can be concisely written as:

``` cpp
this->capacity_ = std::pow(2, std::ceil(std::log2(reserveIndices)));
```